### PR TITLE
Create partial viewers to show data with only structures or only map

### PR DESCRIPTION
**Pull request by ceriottm**
Objective is to have these both as complement to DefaultVisualizer and as jupyter widgets (which is the real goal for me)

### CODE REVIEW

File: python/nbextension/src/widget.ts
Comment (Luthaf) [inline python/nbextension/src/widget.ts:101]: This is not needed for a pure structure viewer

File: python/chemiscope/jupyter.py
Comment (Luthaf) [inline python/chemiscope/jupyter.py:108]:
```suggestion
        return StructureWidget(dict_input)
    else:
        raise ValueError(f"invalid mode '{mode}' in chemiscope.show, expected one of 'default' or 'structure'")
```

File: src/index.ts
Comment (Luthaf) [inline src/index.ts:509]: I don't think we want to have the `saveSettings`, `applySettings`, or `dataset` for StructureVisualizer. They are purely duplicated code which no-one uses, and would be relatively easy to add back if we need them later.

Comment (ceriottm) [reply src/index.ts:509]: I agree for the jupyter use case but how about the object that can be used in a webpage? Don't we want to give the possibility of applying settings? Alternative would be to move these to the chemiscope app and out of the viewers.

Comment (Luthaf) [reply src/index.ts:509]: > how about the object that can be used in a webpage?

Since nobody is using this functionality, I would use YAGNI and remove code. 

Moving the settings functions to the app is also a good idea, I'll do it in a separate PR once this one is merged.

**Comment by ceriottm** [comment]
Ready for review

**Review by jakublala — APPROVED** [review]

**Comment by Luthaf** [comment]
Do we really want to have a metadata panel in these new modes? Since they are intended mainly for use in jupyter where the metadata is often not defined, I think that it does not make sense to display them (they only show a ⓘ symbol).

**Comment by ceriottm** [comment]
I think actually that this shows we should not display the (i) when the
metadata is empty. Again, I can see uses of the "one panel chemiscope" in a
website, so to me it made sense to have a meta panel available.
Michele

**Comment by Luthaf** [comment]
Rebased on top of #198. Should be good to merge after some more testing!

**Pull request merged**
